### ISSUE TITLE
Allow declarations in Faker kwargs

### ISSUE THREAD

**Issue by maxrothman** [FactoryBoy/factory_boy#729]
#### The problem
Take this model for example:

```python
class User:
    address
    city
    state
    zip
```

A first approach to building a factory would look like this:

```python
class User(Factory):
    address = Faker('street_address')
    city = Faker('city')
    state = Faker('state')
    zip = Faker('zipcode')
```

But this could cause spurious bugs since the zip and the state might not match. Instead, one might want to take this approach:

```python
class User(Factory):
    address1 = Faker('street_address')
    city = Faker('city')
    state = Faker('state')
    zip = Faker('zipcode_in_state', state_abbr=SelfAttribute('state'))
```

However this fails with the following error:
```
/usr/local/lib/python3.6/dist-packages/faker/providers/address/en_US/__init__.py:427: in zipcode_in_state
    return self.postcode_in_state(state_abbr)

Exception: State Abbreviation not found in list
```

After doing some digging, it appears that the `SelfAttribute` is not being evaluated before being passed to Faker.

#### Proposed solution
Declarations should be evaluated before being passed to faker, the same way they are before being passed to subfactories.

**Comment by tasawar-hussain** [comment]
use LazyAttributes: 
https://github.com/FactoryBoy/factory_boy#lazy-attributes

**Comment by maxrothman** [comment]
You mean like `zip = LazyAttribute(lambda o: Faker('zipcode_in_state', state_abbrev=o.state))`? Won't that result in my model getting called with `Model(zip=<Faker declaration>`?

**Comment by interDist** [comment]
@maxrothman I believe you should use `Faker(...).generate({})` in the LazyAttribute. See mine #732.

**Comment by maxrothman** [comment]
@interDist ah, I see what you mean. Looking at #732, it appears to describe a similar issue to this one. I appreciate the workaround, but would y'all be interested in the proposed fix?

**Comment by rbarrois** [comment]
Hi!

Thanks for the report!

This has been covered by #772, and should be available in release 3.1 ;)
